keyring-util/src/main.rs

```diff
@@ -0,0 +1,259 @@
+impl From<oo7::Error> for Error {
+    fn from(_err: oo7::Error) -> Error {
+        // TODO
+        Error(String::new())
+    }
+}
+
+impl Error {
+    fn new(s: &str) -> Self {
+        Self(String::from(s))
+    }
+}
+
+impl Termination for Error {
+    fn report(self) -> ExitCode {
+        ExitCode::FAILURE
+    }
+}
+
+#[async_std::main]
+async fn main() -> Result<(), Error> {
+    let cmd = Command::new("keyring-util")
+        .bin_name("keyring-util")
+        .subcommand_required(true)
+        .subcommand(
+            Command::new("store")
+                .arg(
+                    arg!(label: <LABEL>)
+                        .required(true)
+                        .help("Description for the secret")
+                        .value_parser(value_parser!(String)),
+                )
+                .arg(
+                    arg!(attributes: [ATTRIBUTES] ...)
+                        .help("List of attributes. This is a space separated list of pairs of key value")
+                        .value_parser(value_parser!(String))
+                )
+                .after_help("The contents of the secret will be asked afterwards.\n\nExample:\n  keyring-util store 'My Personal Mail' smtp-port 1025 imap-port 143")
+                .about("Store a secret"),
+        )
+        .subcommand(
+            Command::new("lookup")
+                .arg(
+                    arg!(attributes: [ATTRIBUTES] ...)
+                        .help("List of attributes. This is a space separated list of pairs of key value")
+                        .value_parser(value_parser!(String))
+                )
+                .after_help("Example:\n  keyring-util lookup smtp-port 1025")
+                .about("Retrieve a secret"),
+        )
+        .subcommand(
+            Command::new("delete")
+                .arg(
+                    arg!(attributes: [ATTRIBUTES] ...)
+                        .help("List of attributes. This is a space separated list of pairs of key value")
+                        .value_parser(value_parser!(String))
+                )
+                .after_help("Will delete all secrets with matching attributes.\n\nExample:\n  keyring-util delete smtp-port 1025")
+                .about("Delete a secret"),
+        )
+        .subcommand(
+            Command::new("search")
+                .about("Search entries with matching attributes")
+                .after_help("Example:\n  keyring-util search --all smtp-port 1025")
+                .arg(
+                    arg!(--"all")
+                        .help("Whether to list all possible matches or only the first result")
+                )
+                .arg(
+                    arg!(attributes: [ATTRIBUTES] ...)
+                        .help("List of attributes. This is a space separated list of pairs of key value")
+                        .value_parser(value_parser!(String))
+                ),
+        )
+        .subcommand(Command::new("lock").about("Lock the keyring"))
```

Comment:
I think the new `lock` and `unlock` commands are nice. But, do we really need them here in this tool? I'm asking this because, `secret-tool` only does "store and retrieve passwords" and not lock/unlock keyring. For me personally I always prefer to keep my keyring(s) in a locked state and avoid accidental lock/unlock situations :)

(will update my review after seeing the implementation of this).

Reply:
I would keep them for backward compatibility purposes